### ISSUE TITLE
Tags aren't working

### ISSUE THREAD

**Issue by jakub-kolosowski-pirum** [amiceli/vitest-cucumber#163]
Hi! 
We have encountered that even though you have unit tests for tags/filtering we faced the problem that it runs all of them every time. 
Version installed: 4.1.2
We followed documentation so put custom tag before Scenario -> @ignore
and then we added in describeFeature excludeTags and all the tests are still being triggered.
We tried also using configuration - setVitestCucumberConfiguration but it is not working either.
Is it a bug or we are missing something that's not included in docs?

**Comment by amiceli** [comment]
Hi,
`@ignore` is already use in default option, you don't have to add it.

Do you have excluded other tags ?
When you run test, you have error like "Scenario test is not called" ?

**Comment by jakub-kolosowski-pirum** [comment]
I did have used other tags to exclude than @ignore. Answering your second question, no I don't have that error, just all tests are passing like running without tags.

**Comment by amiceli** [comment]
Can you show some code please.
I think I found a bug, can you test to add "ignore" in `excludeTags` where you added other tag.

Example, replace `excludeTags: ["ignore-e2e"]` with `excludeTags: ["ignore-e2e", "ignore"]`.

**Comment by jakub-kolosowski-pirum** [comment]
So I have added @ignore tag to my scenario:
```
Feature: XYZ feature

  @ignore
  Scenario: XYZ scenario
    Given XYZ given
    When XYZ when
    Then XYZ then
    And XYZ and
```
and put in describe feature in steps
```
const feature = await loadFeature("features/index.feature");

describeFeature(
  feature,
  ({ Scenario }) => {
    Scenario(
      "XYZ scenario",
      ({ Given, When, Then, And }) => {
        Given(
          "XYZ given",
          () => {}
        );

        When("XYZ when", () => { });

        Then("XYZ then", () => {});

        And("XYZ and", () => {});
      });
    },
    { excludeTags: ["ignore", "ignore-e2e"] }
  );
  ```
  still the test is being run

**Comment by jakub-kolosowski-pirum** [comment]
On top of that when I change ignore to something different for example like in docs `awesome` it's still running that test and passes

**Comment by amiceli** [comment]
Ok, I'm going to check this and I'll made a fix.

**Comment by Odonno** [comment]
I tried that with the Playground we have in the docs website. Adding the `@ignore` tag indeed does nothing.

@amiceli It should not throw an error but execute `describe.skip` if an exclusion tag is present.

**Comment by amiceli** [comment]
This is a bug, I will fix it on Playground too.
No, when you ignore a Scenario, it shouldn't be implemented. It's like keep useless code in a project.

I will throw en error in fix but you can remove `@ignore` tag and use keep `skip` in step if you want to keep scenario test code.

An another idea, I can implement a `@skip` tag, to skip Scenario/Rule/Background tests.

**Comment by Odonno** [comment]
No, it should not throw. This is not what developers would expect. This is idea is to able to avoid running some Scenario. But code should not be removed because it would require too much effort just to run the tests without some Scenarii. See the original implementation and documentation here: https://github.com/bencompton/jest-cucumber/blob/main/docs/AdditionalConfiguration.md#tag-filtering

However, I can agree with you in a sense that only want to exclude in a sense you'd want to skip those scenarii. If you want to have a proper naming convention, one that does not give confusion, then that could be an idea. 

But then, I don't see the need of `excludeTags` as it would bring confusion with this new idea. That would mean to do:

* rename `excludeTags` by `skipTags` and then have a default value of `@skip` (instead of the current `@ignore`)
* if we go that way, why not also `todoTags` with a default value of `@todo` to mark unfinished scenarii (and steps?)

**Comment by amiceli** [comment]
Ok now I understand. I'm ok to call `skip` when a scenario should be ignored ;).
I will make a fix for this.

**Comment by francesco-albanese** [comment]
I think that `excludeTags` and `includeTags` would still be very useful, especially when the same batch of end to end tests can be executed differently based on requirements. For example the tag `@smoke` can be used only to run scenarios tagged with `@smoke` , by a CI to do regression tests. And the tags should be specified as an environment variable so that can be controlled from a CI environment.

**Comment by francesco-albanese** [comment]
Yes that would work very well. But most important thing is being able to support scenario filtering based by tags, just like jest cucumber and similar libraries. Using tags in features files is a common practice for scenario filtering and it's even described in the official cucumber documentation https://cucumber.io/docs/cucumber/api/?lang=javascript

**Comment by amiceli** [comment]
Ok for env variable, I will work on another PR for it.
For tags currently vitest-cucumber provides `includeTags` and `excludeTags`.

On the fix I will call `skip` for scenarios not matching `includeTags`.

**Comment by amiceli** [comment]
I've published a beta version `4.1.3-beta.1`.
You can test it. I'm making some tests with large projects.

**Comment by jakub-kolosowski-pirum** [comment]
Amazing @amiceli! It works properly with any tags now for exclude and include also in configuration. Thank you so much!

**Comment by amiceli** [comment]
Nice, thanks for your tests ;).
I'm going to publish a stable version.

**Comment by amiceli** [comment]
I published a stable version `4.1.3`.
I will update doc about tags, `skip` etc.

@francesco-albanese I will open another issue for env variables.
